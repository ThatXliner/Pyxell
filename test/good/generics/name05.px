
func call1<A,B,C>(A->B f, A x) C def
    A x' = x
    B r = f(x')
    return r

func call2<A,B,C>(B x, B->A f) C def
    B->A f' = f
    r = call1(f', x)
    A r' = r
    return r'

print call2(4.5, _.toString())
print call1(_.toString(), null)
print call2(['$'], _[0])
