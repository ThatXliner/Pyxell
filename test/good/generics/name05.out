4.5
null
$
